### PR TITLE
CPLAT-8834 fix tests for redux 7.1.3

### PR DESCRIPTION
**Pull request by kealjones-wk**
## Motivation
  
The NPM package `react-redux` that over_react consumes from react-dart [changed its "Provider" component from a class based component to a function component](https://github.com/reduxjs/react-redux/pull/1377) in the 7.1.1 __patch__ version. This broke some expectations that our tests had in place for over_react_redux.

## Changes
  
- Remove usage of Class only features from over_react_redux Provider tests  

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
      - [ ] open react-dart and update `packages.json`'s `react-redux` version to `"7.1.3"`
      - [ ] run `npm run build` and publink react-dart to over_react.
      - [ ] run `pub run dart_dev test -P dartdevc` and make sure all tests pass.
      - [ ] run `pub run dart_dev test --build-args="-r" -P dart2js` and make sure all tests pass.
        
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/over_react/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: test/over_react_redux/connect_test.dart [hunk 1]
```diff
@@ -29,7 +29,7 @@ main() {
   group('connect', () {
     UiFactory<CounterProps> ConnectedCounter;
     TestJacket<CounterComponent> jacket;
-    dynamic counterRef;
+    Ref<CounterComponent> counterRef = createRef();
```
Comment (greglittlefield-wf) [inline test/over_react_redux/connect_test.dart]: #nit could be final
```suggestion
    final counterRef = createRef<CounterComponent();
```

File: test/over_react_redux/connect_test.dart [hunk 2]
```diff
@@ -268,44 +268,44 @@ main() {
...
+        expect(counterRef.current.props.currentCount, 1);
+        expect(jacket.mountNode.innerHtml, contains('Count: 1'));
       });
     });
 
     group('Options', () {
       group('areOwnPropsEqual', () {
         test('', () {
           ConnectedCounter = connect<CounterState, CounterProps>(
-            areOwnPropsEqual: expectAsync2((next, prev) {
```
Comment (greglittlefield-wf) [inline test/over_react_redux/connect_test.dart]: Why was this removed?

File: test/over_react_redux/connect_test.dart [hunk 3]
```diff
@@ -353,20 +352,20 @@ main() {
...
           expect(methodsCalled, ['mapStateToProps', 'areStatePropsEqual']);
-          expect(jacket.getNode().innerHtml, contains('Count: 0'));
+          expect(jacket.mountNode.innerHtml, contains('Count: 0'));
         });
       });
 
       group('areMergedPropsEqual', () {
         test('', () {
           ConnectedCounter = connect<CounterState, CounterProps>(
-            areMergedPropsEqual: expectAsync2((next, prev) {
```
Comment (greglittlefield-wf) [inline test/over_react_redux/connect_test.dart]: Same question here

File: test/over_react_redux/connect_test.dart (hunk 1 above)
Comment (kealjones-wk) [reply test/over_react_redux/connect_test.dart]:
```suggestion
    final counterRef = createRef<CounterComponent>();
```

File: test/over_react_redux/connect_test.dart [hunk 4]
```diff
@@ -328,23 +328,22 @@ main() {
             areStatePropsEqual: (next, prev) {
               expect(next, isA<CounterProps>());
               expect(prev, isA<CounterProps>());
-              expect(next.currentCount, 1);
               methodsCalled.add('areStatePropsEqual');
-              // Force it to always be true, meaing it shouldnt re-render if they change.
+              // Force it to always betrue, meaing it shouldnt re-render if they change.
```
Comment (aaronlademann-wf) [inline test/over_react_redux/connect_test.dart]:
```suggestion
              // Force it to always be true, meaing it shouldnt re-render if they change.
```

File: test/over_react_redux/connect_test.dart (hunk 2 above)
Comment (kealjones-wk) [reply test/over_react_redux/connect_test.dart]: honestly, because i added somewhere else and i forgot which one it was that i added and ended up removing them all lol ill add these back.

File: test/over_react_redux/connect_test.dart (hunk 1 above)
Comment (greglittlefield-wf) [reply test/over_react_redux/connect_test.dart]: :shame:

File: test/over_react_redux/connect_test.dart [hunk 5]
```diff
@@ -401,7 +401,8 @@ main() {
             ),
           );
 
-          expect(methodsCalled, ['mapStateToProps', 'areStatesEqual']);
```
Comment (greglittlefield-wf) [inline test/over_react_redux/connect_test.dart]: FYI you can also do 
```dart
containsAll(['mapStateToProps', 'areStatesEqual']));
```

File: test/over_react_redux/connect_test.dart [hunk 6]
```diff
@@ -398,24 +397,24 @@ main() {
...
-              (ConnectedCounter()..ref = (ref){ counterRef = ref; }..currentCount = 0)('test'),
+              (ConnectedCounter()..ref = counterRef..currentCount = 0)('test'),
             ),
           );
 
-          // `mapStateToProps` is called once,
-          // then `areStatesEqual` shows up 2 times due to `initialState`.
-          expect(methodsCalled, ['mapStateToProps', 'areStatesEqual', 'areStatesEqual']);
+          expect(methodsCalled, contains('mapStateToProps'));
+          expect(methodsCalled, contains('areStatesEqual'));
```
Comment (greglittlefield-wf) [inline test/over_react_redux/connect_test.dart]: FYI you can also do 
```dart
containsAll(['mapStateToProps', 'areStatesEqual']));
```
